packages/text-annotator/src/state/spatialTree.ts

```diff
@@ -19,35 +23,40 @@ export interface IndexedHighlightRect {
 
   maxY: number;
 
-  annotation: {
-    
+  target: {
+
     id: string;
 
+    annotation: string;
+
     rects: DOMRect[];
 
-  }
+  };
```

Comment:
Now the `target` is the main entity for the indexed highlight rects. Because each `target` has its own set of `rects` and each annotation can have _multiple targets_